db_classes1.py

```diff
@@ -121,36 +124,92 @@ def __init__(self, department_id, employee_id, database):
+                                    JOIN DEPARTMENT_EMPLOYEE ON EMPLOYEE.ID=DEPARTMENT_EMPLOYEE.employee_id
+                                    WHERE DEPARTMENT_EMPLOYEE.department_id = {self.department_id}
+                                    AND HISTORY_VACATION.finish_date > DATE('now')'''
+        if self.database.execute_sql(employees_on_vacation).fetchone()[0] > 5:
+            raise QuantityLimit('Currently on vacation are 5 employees')
         return self.database.save(self.sql_script)
 
     def get(self, *args, **kwargs):
-        self.sql_script = f'''SELECT {args} FROM department'''
+        self.sql_script = f'''SELECT {args} FROM HistoryVacation'''
         return self.database.get(self.sql_script)
 
-    def check_number_employees_vacation(self):
-        pass
-
 
-db = Database('factory')
+class Company:
+    def __init__(self):
+        self.list_departments = []
+        self.list_employees = []
+        self.department_employee = {}
+        self.employees_vacation = []
+
+    def add_department(self, id, name, abbreviation, max_amount, database):
+        department = Department(id, name, abbreviation, max_amount, database)
+        department.save()
+        self.list_departments.append(department)
+
+    def add_employee(self, id, employment_date, fired_date, head_officer, person_id, position_id, id_person, first_name,
+                     second_name, middle_name, passport, birthday, place_birthday, address, database, position):
+        employee = Employee(id, employment_date, fired_date, head_officer, person_id, position_id, id_person, first_name,
+                            second_name, middle_name, passport, birthday, place_birthday, address, database, position)
+        employee.save()
+        self.list_employees.append(employee)
+
+    def get_salary(self, employee):
+        for emp in self.list_employees:
+            if emp == employee:
+                experience = employee.employment_date - employee.fired_date
+                if experience >= 1:
+                    return employee.position.rate*experience*1.012
+                return employee.position.rate
+
+    def allocation_employee_department(self, employee_id, department_id, database):
+        emp_to_dep = DepartmentEmployee(employee_id, department_id, database)
+        try:
+            emp_to_dep.save()
+        except QuantityLimit as q:
+            print(q)
+        else:
+            self.department_employee[department_id] = employee_id
+
+    def take_vacation(self, id, start_date, finish_date, employee_id, department_id, database):
+        vacation = HistoryVacation(id, start_date, finish_date, employee_id, department_id, database)
+        try:
+            vacation.save()
+        except QuantityLimit as q:
+            print(q)
+        else:
+            self.employees_vacation.append(employee_id)
+
+
+class QuantityLimit(Exception):
+    def __init__(self, arg):
```

Comment:
For `Exception` inheritance you should call `Exception` constructor. As for usual inheritance.
For `Exception` hierarchy it's enough to just inherit without additional functionality
`class QuantityLimit(Exception): pass`